src/Task.cc

```diff
@@ -886,6 +888,23 @@ void Task::activate_preload_thread_locals() {
   }
 }
 
+static bool cpu_has_singlestep_quirk() {
+  static uint8_t has_quirk = 0xff;
```

Comment:
Actually to simplify testing and maintenance, if this code doesn't cause a significant performance penalty in practice (and I don't see why it should), I think we should just do this for all CPU models.